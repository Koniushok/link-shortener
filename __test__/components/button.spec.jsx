import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Button from '../../src/components/button';
import 'jest-styled-components';

describe('Button styled component', () => {
  describe('Button component initial', () => {
    it('render correctly component', () => {
      const output = shallow(<Button />);
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });
  describe('Button component render margin', () => {
    it('render alignRight margin', () => {
      const button = mount(<Button alignRight />);
      expect(button).toHaveStyleRule('margin', '0 0 0 auto');
    });
    it('render alignLeft margin', () => {
      const button = mount(<Button alignLeft />);
      expect(button).toHaveStyleRule('margin', '0 auto 0 0');
    });
    it('render alignCenter margin', () => {
      const button = mount(<Button alignCenter />);
      expect(button).toHaveStyleRule('margin', '0 auto');
    });
    it('render default margin', () => {
      const button = mount(<Button />);
      expect(button).toHaveStyleRule('margin', '0');
    });
  });
});
