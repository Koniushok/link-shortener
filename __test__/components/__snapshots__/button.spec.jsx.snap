// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styled component Button component initial render correctly component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: block;
  padding: 9px 15px;
  font-size: 20px;
  font-weight: 600;
  border-radius: 6px;
  color: #fff;
  background: #6c757d;
  border: solid 3px #6c757d;
  margin: ",
          [Function],
          ";
  cursor: pointer;
  :focus {
    border: 3px solid #a9a9a9;
    outline: #a9a9a9;
  }
  :hover {
    border: 3px solid #ffffff9c;
  }
  :disabled {
    background: #d9d9d9;
    border: 3px solid #ffffff9c;
    cursor: default;
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
